dl/src/main/scala/com/intel/analytics/bigdl/nn/Recurrent.scala

```diff
@@ -29,83 +29,172 @@ class Recurrent[T : ClassTag] (
   bpttTruncate: Int = 2)
   (implicit ev: TensorNumeric[T]) extends Container[Tensor[T], Tensor[T], T] {
 
-  val hidden = Tensor[T]()
-  var module: Module[T] = _
-  var transform: Module[T] = _
+  var hidden: Activity = null
+  val inputs = T()
+  val batchDim = 1
+  val timeDim = 2
+  val hidDim = 1
+  val inputDim = 2
   var (batchSize, times) = (0, 0)
 
+  override def reset(): Unit = {
+    modules(0).reset()
+    modules(0).isInstanceOf[RnnCell[T]] match {
+      case true => hidden = Tensor[T]()
+      case _ => throw new IllegalArgumentException("Cell not implemented")
+    }
+  }
+
+  /**
+   * Clone N models; N depends on the time dimension of the input
+   * @param size
+   */
+  private def extend(size: Int): Unit = {
+    require(modules != null && modules.length >= 1,
```

Comment:
Currently, the Recurrent container only accepts one cell. And each cell should be of trait type Cell (LSTM, RnnCell, GRU, should extends Cell). The multi-layer RNN will be supported. Then there will be a private vairable to store all cloned modules instead of using modules[ArrayBuffer].